components/constellation/constellation.rs

```diff
@@ -2819,7 +2822,19 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
 
         debug!("Closed browsing context children {}.", browsing_context_id);
     }
-
+    
+    // Discard the pipeline for a given document, udpdate the joint session history.
+    fn handle_discard_document(&mut self, top_level_browsing_context_id: TopLevelBrowsingContextId, pipeline_id: PipelineId) {
```

Comment:
@cbrewster I wasn't completely sure what you meant with https://github.com/servo/servo/pull/20329#discussion_r180462129, so I gave it a try here. Please let me know what you think.